traversal/selector/exploreRecursive.go

```diff
@@ -149,8 +154,11 @@ func (s ExploreRecursive) replaceRecursiveEdge(nextSelector Selector, replacemen
 	return nextSelector
 }
 
-// Decide always returns false because this is not a matcher
+// Decide if a node directly matches
 func (s ExploreRecursive) Decide(n ipld.Node) bool {
+	if s.stopAt != nil && s.stopAt.Match(n) {
+		return false
+	}
```

Comment:
I'm wobbling on if this is going to be the desired behaviour.  The semantic question is: does `stopAt` mean "don't explore beyond this", or that _and also_ "don't match this"?

Reply:
I lean towards thinking the second part is unnecessary.

Reasons:
- "stopAt" to me reads as "i will reach this point and go no further".  That means we can still match here.
- Less dropping the "and also don't match this" makes the overall description of behavior shorter.  Shorter is a heuristic for better.
- It's only even relevant if you have a matcher clause that lands on the same node as the stop.  You can just... choose not to do that?
- ... and since the main practical purpose of the stops is to avoid _loading_ more data, usually, then, matching on some data you've already loaded isn't really the big worry.

In that case it means we don't need the new branch here.

I wouldn't mind a second opinion on this, though.  These reasons aren't strongly held.

Reply:
removed